application/application.go

```diff
@@ -13,6 +13,8 @@ type Application interface {
 	SpaceResources() space.ResourceRepository
 	Users() account.UserRepository
 	OauthStates() auth.OauthStateReferenceRepository
+	ExternalProviderTokens() token.ExternalProviderTokenRepository
+	ExternalProviders() token.ExternalProviderRepository
```

Comment:
I wonder if we really need a token provider table for now... Right now we are going to support github & OSO providers only. For OSO it's just starter-us-east-2.openshift.com.
For provider we need:
- type
- URL
- client ID
- client Secret

So, for now we could just hardcode two providers (configured via configmaps/secrets) for GitHub and OSO east-2. By the time we need to support multiple OSO clusters we most likely won't need to link OSO accounts at all (we will use proxy for accessing OSO). So, the only provider left will be GitHub.
Later, when we are ready to support other providers we will need to introduce some sort of Provider manager. Then we most likely will need a Provider table.

WDYT?

Reply:
Agreed, we can have all the github & OSO creds available as configuration.
In our `ExternalProviderToken` table, instead of provider ID, we can store provider type as a string. And all info related to provider type should be looked up in configuration.

Reply:
Started using `ProviderID`  https://github.com/fabric8-services/fabric8-auth/pull/131/commits/f74da6bc9bf1edc65959b85f29bc79728b25b1be and removed `ExternalProviders`